src/interfaces/swig/factory.i

```diff
@@ -61,28 +87,31 @@ namespace shogun{
-%template(create_distribution) shogun::create<shogun::Distribution, std::string>;
-%template(create_combination_rule) shogun::create<shogun::CombinationRule, std::string>;
-%template(create_distance) shogun::create<shogun::Distance, std::string>;
-%template(create_kernel) shogun::create<shogun::Kernel, std::string>;
-%template(create_features) shogun::create<shogun::Features, std::string>;
-%template(create_machine) shogun::create<shogun::Machine, std::string>;
-%template(create_structured_model) shogun::create<shogun::StructuredModel, std::string>;
-%template(create_factor_type) shogun::create<shogun::FactorType, std::string>;
+%template(create_svm) shogun::create_<shogun::SVM,std::string>;
+%template(create_evaluation) shogun::create_<shogun::Evaluation, std::string>;
+%template(create_multiclass_strategy) shogun::create_<shogun::MulticlassStrategy, std::string>;
+%template(create_ecoc_encoder) shogun::create_<shogun::ECOCEncoder, std::string>;
+%template(create_ecoc_decoder) shogun::create_<shogun::ECOCDecoder, std::string>;
+%template(create_transformer) shogun::create_<shogun::Transformer, std::string>;
+%template(create_layer) shogun::create_<shogun::NeuralLayer, std::string>;
+%template(create_splitting_strategy) shogun::create_<shogun::SplittingStrategy, std::string>;
+%template(create_machine_evaluation) shogun::create_<shogun::MachineEvaluation, std::string>;
+%template(create_gp_likelihood) shogun::create_<shogun::LikelihoodModel, std::string>;
+%template(create_gp_mean) shogun::create_<shogun::MeanFunction, std::string>;
+%template(create_gp_inference) shogun::create_<shogun::Inference, std::string>;
+%template(create_differentiable) shogun::create_<shogun::DifferentiableFunction, std::string>;
+%template(create_loss) shogun::create_<shogun::LossFunction, std::string>;
+%template(create_parameter_observer) shogun::create_<shogun::ParameterObserver, std::string>;
+%template(create_evaluation_result) shogun::create_<shogun::EvaluationResult, std::string>;
+%template(create_distribution) shogun::create_<shogun::Distribution, std::string>;
+%template(create_combination_rule) shogun::create_<shogun::CombinationRule, std::string>;
+%template(create_distance) shogun::create_<shogun::Distance, std::string>;
+%template(create_kernel) shogun::create_<shogun::Kernel, std::string>;
+%template(create_features) shogun::create_<shogun::Features, std::string>;
+%template(create_machine) shogun::create_<shogun::Machine, std::string>;
+%template(create_structured_model) shogun::create_<shogun::StructuredModel, std::string>;
+%template(create_factor_type) shogun::create_<shogun::FactorType, std::string>;
+%template(create_gaussian_process) shogun::create_<shogun::GaussianProcess, std::string>;
 %template(create_labels) shogun::create_labels<float64_t>;
+%template(create_minimizer) shogun::create_<shogun::Minimizer, std::string>;
+%template(create_lbfgs_minimizer) shogun::create_<shogun::LBFGSMinimizer, std::string>;
```

Comment:
Do you need to have this? Isn't this a subclass of Minimizer?

Reply:
this method is used in many place.
https://github.com/shogun-toolbox/shogun/blob/e8a2d2068cbcf6d83ff6c4fe2bea2206060c6a97/src/shogun/optimization/lbfgs/LBFGSMinimizer.h#L110-L126

Reply:
Hmm, I think this has to be refactored. You mean this is called from inside the examples? Because these values should be updated with `put` in the interfaces

Reply:
might be good to add that in this PR actually, as then we don't need to have this in swig? feasible? The PR is already quite big ...

Reply:
@LiuYuHui I think this is the last thing that has be worked on